lib/identity_job_log_subscriber.rb

```diff
@@ -0,0 +1,168 @@
+# ActiveJob events documentation:
+# https://edgeguides.rubyonrails.org/active_support_instrumentation.html#active-job
+# https://github.com/rails/rails/blob/v6.1.3.1/activejob/lib/active_job/log_subscriber.rb
+class IdentityJobLogSubscriber < ActiveSupport::LogSubscriber
+  def enqueue(event)
+    job = event.payload[:job]
+    ex = event.payload[:exception_object]
+
+
+    json = default_attributes(event, job)
+
+    if ex
+      json[:exception_class] = ex.class
+      json[:exception_message] = ex.message
+
+      error(json.to_json)
+    elsif event.payload[:aborted]
+      info(json.to_json)
+    else
+      json[:job_id] = job.job_id
+
+      info(json.to_json)
+    end
+  end
+
+  def enqueue_at(event)
+    job = event.payload[:job]
+    ex = event.payload[:exception_object]
+
+    json = default_attributes(event, job)
+
+    if ex
+      json[:exception_class] = ex.class
+      json[:exception_message] = ex.message
+
+      error(json.to_json)
+    elsif event.payload[:aborted]
+      json[:halted] = true
+
+      info(json.to_json)
+    else
+      json[:scheduled_at] = scheduled_at(event)
+      json[:job_id] = job.job_id
+
+      info(json.to_json)
+    end
+  end
+
+  def perform_start(event)
+    job = event.payload[:job]
+
+    json = default_attributes(event, job).merge(
+      job_id: job.job_id,
+      enqueued_at: job.enqueued_at,
+      queued_duration_ms: queued_duration(job),
+    )
+
+    info(json.to_json)
+  end
+
+  def perform(event)
+    job = event.payload[:job]
+    ex = event.payload[:exception_object]
+
+    json = default_attributes(event, job).merge(
+      job_id: job.job_id,
+      enqueued_at: job.enqueued_at,
+      queued_duration_ms: queued_duration(job),
+    )
+
+    if ex
+      # NewRelic?
+      json[:exception_class] = ex.class
+      json[:exception_message] = ex.message
+      json[:exception_backtrace] = Array(ex.backtrace).join("\n")
+
+      error(json.to_json)
+    elsif event.payload[:aborted]
+      json[:halted] = true
+
+      error(json.to_json)
+    else
+      info(json.to_json)
+    end
+  end
+
+  def enqueue_retry(event)
+    job = event.payload[:job]
+    ex = event.payload[:error]
+    wait_seconds = event.payload[:wait]
+
+    json = {
+      job_id: job.job_id,
+      wait_ms: wait.to_i.in_milliseconds
+    }
+
+    if ex
+      json[:exception_class] = ex.class
+    end
+
+    info(json.to_json)
+  end
+
+  def retry_stopped(event)
+    job = event.payload[:job]
+    ex = event.payload[:error]
+
+    json = default_attributes(event, job).merge(
+      job_id: job.job_id,
+      exception_class: ex.class,
+      attempts: job.executions,
+    )
+
+    error(json.to_json)
+  end
+
+  def discard(event)
+    job = event.payload[:job]
+    ex = event.payload[:error]
+
+    json = default_attributes(event, job).merge(
+      job_id: job.job_id,
```

Comment:
WDYT of adding `job_id` to `default_attributes`?

Reply:
good call, moved it to default. It initially didn't seem like `job_id` existed in the `enqueue` event based on how Rails used it, but it is there.